src/frontend/src/optimizer/plan_node/logical_multi_join.rs

```diff
@@ -111,6 +114,167 @@ impl PlanTreeNode for LogicalMultiJoin {
     }
 }
 
+impl LogicalMultiJoin {
+    pub fn as_reordered_left_deep_join(&self, join_ordering: &[usize]) -> PlanRef {
+        assert_eq!(join_ordering.len(), self.inputs.len());
+        assert!(!join_ordering.is_empty());
+
+        let base_plan = self.inputs[join_ordering[0]].clone();
+
+        // Express as a cross join, we will rely on filter pushdown to push all of the join
+        // conditions to convert into inner joins.
+        let mut output = join_ordering[1..]
+            .iter()
+            .fold(base_plan, |join_chain, &index| {
+                LogicalJoin::new(
+                    join_chain,
+                    self.inputs[index].clone(),
+                    JoinType::Inner,
+                    Condition::true_cond(),
+                )
+                .into()
+            });
+
+        if join_ordering != (0..self.input_col_nums().iter().sum()).collect::<Vec<_>>() {
+            output =
+                LogicalProject::with_mapping(output, self.mapping_from_ordering(join_ordering))
```

Comment:
so this projection is only used for reordering columns? I heard that someone is working on `output_index`, maybe can be applied here... cc @st1page

Reply:
yes, after this PR merged.

Reply:
Yes, we could simply put the output_index on the final join?

Reply:
Yes for now, but maybe could be better after every binary join support output index.